hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AbfsConfiguration.java

```diff
@@ -635,4 +641,24 @@ void setWriteBufferSize(int bufferSize) {
   void setEnableFlush(boolean enableFlush) {
     this.enableFlush = enableFlush;
   }
+
+  private String getMsiAuthEndpoint() throws IOException {
+    String authEndpoint = getPasswordString(
+        FS_AZURE_ACCOUNT_OAUTH_MSI_ENDPOINT);
+    if (Utils.isEmpty(authEndpoint)) {
+      authEndpoint =
+          AuthConfigurations.DEFAULT_FS_AZURE_ACCOUNT_OAUTH_MSI_ENDPOINT;
+    }
+    return authEndpoint;
+  }
+
+  private String getRefreshTokenAuthEndpoint() throws IOException {
+    String authEndpoint = getPasswordString(
+        FS_AZURE_ACCOUNT_OAUTH_REFRESH_TOKEN_ENDPOINT);
+    if (Utils.isEmpty(authEndpoint)) {
```

Comment:
minor: this new function isEmpty() might not be necessary

Reply:
Done